libminifi/include/io/AsioStream.h

```diff
@@ -0,0 +1,92 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <string>
+#include <memory>
+#include <utility>
+
+#include "BaseStream.h"
+#include "core/logging/LoggerFactory.h"
+#include "asio/ts/internet.hpp"
+#include "asio/read.hpp"
+#include "asio/write.hpp"
+#include "io/validation.h"
+
+namespace org::apache::nifi::minifi::io {
+
+template<typename AsioSocketStreamType>
+class AsioStream : public io::BaseStream {
+ public:
+  explicit AsioStream(AsioSocketStreamType&& stream) : stream_(std::move(stream)) {}
+
+  /**
+   * Reads data and places it into buf
+   * @param buf buffer in which we extract data
+   * @param buflen
+   */
+  size_t read(std::span<std::byte> buf) override;
+
+  /**
+   * writes value to stream
+   * @param value value to write
+   * @param size size of value
+   */
```

Comment:
This comment is misleading; I would remove it and also change the name of the first parameter, since it points to the start of a buffer, not to a single value.

The parameter names could e.g. be `source_buffer` here, and `target_buffer` in `read()`.

Reply:
Updated parameter names in 45c677990eeb0cb223814b3c55d37e84711a9b03

Reply:
It's the other way round: `read()` reads from the stream into the target buffer, and `write()` writes the contents of the source buffer to the stream.

Reply:
Yes, my mistake, fixed in 055547275fb03b502d81d1f505c49656eb250bc0